src/LoadBalancer.ts

```diff
@@ -0,0 +1,127 @@
+import { KDPoint, KDTree, Logger } from 'edumeet-common';
+import MediaNode from './media/MediaNode';
+import { Peer } from './Peer';
+import Room from './Room';
+import * as geoip from 'geoip-lite';
+
+const logger = new Logger('LoadBalancer');
+
+export interface LoadBalancerOptions {
+	kdTree: KDTree;
+	defaultClientPosition: KDPoint;
+	cpuLoadThreshold?: number;
+	geoDistanceThreshold?: number;
+}
+
+export default class LoadBalancer {
+	private readonly cpuLoadThreshold: number;
+	private readonly geoDistanceThreshold: number;
+	private kdTree: KDTree;
+	private readonly defaultClientPosition: KDPoint;
+
+	constructor({
+		kdTree,
+		defaultClientPosition,
+		cpuLoadThreshold = 0.85,
+		geoDistanceThreshold = 2000
+	}: LoadBalancerOptions) {
+		logger.debug('constructor() [kdTree: %s, defaultClientPosition: %S]', kdTree, defaultClientPosition);
+		this.kdTree = kdTree;
+		this.defaultClientPosition = defaultClientPosition;
+		this.cpuLoadThreshold = cpuLoadThreshold;
+		this.geoDistanceThreshold = geoDistanceThreshold;
+	}
+
+	public getCandidates(
+		room: Room,
+		peer: Peer
+	): MediaNode[] {
+		try {
+			logger.debug('getCandidates() [room.id: %s, peer.id: %s]', room.id, peer.id);
+			let candidates: MediaNode[] = [];
+
+			// Get sticky candidates
+			candidates = room.getActiveMediaNodes();
```

Comment:
This would fit nicely into a one-liner:
```
let candidates = room.getActiveMediaNodes()
    .filter((m) => m.load < this.cpuLoadThreshold)
    .sort((a, b) => a.load - b.load);
```

Reply:
Seems you forgot to remove some code

Reply:
Indeed. Fixed.